### ISSUE TITLE
Broken type reference link

### ISSUE THREAD

**Issue by beyang** [sourcegraph/srclib-java#82]
1. Go to https://sourcegraph.com/github.com/uber/rides-android-sdk/-/def/JavaArtifact/com.uber.sdk/core-android/-/com/uber/sdk/android/core/auth/LoginActivity:type/sessionConfiguration
2. Note that "SessionConfiguration" right before "sessionConfiguration" is a broken link.

![image](https://cloud.githubusercontent.com/assets/1646931/16314381/59321e9e-3932-11e6-9191-d9a994f7f5d1.png)

**Comment by alexsaveliev** [comment]
The reason is a combination of two issues:

- incorrect target resolution (addressed in alexsaveliev/srclib-java@ed77e5bbbfa10e0c7ae336ed6dde25cfa35c43e3)
- different artifact names: `rides-android-sdk` refers to `com.uber.sdk/rides` while `github.com/uber/rides-java-sdk` provides `com.uber.sdk/sdk` (`com.uber.sdk/rides` is an umbrella artifact there). I am not sure we can fix it short term

**Comment by beyang** [comment]
Got it, is it a configuration error on their part?

**Comment by alexsaveliev** [comment]
No, it's rather unsupported feature on our side. They have umbrella artifact `rides` which includes `sdk` and others. We detecting presence of `sdk` project but not `rides` one because it lists no source files.
Maybe at some point we should add "compound" units to srclib which would try to resolve ref to any of enclosed unit's ref but now I don't have clear vision how to deal with such configurations.
